### ISSUE TITLE
Ability to close the popup window

### ISSUE THREAD

**Issue by mebibou** [720kb/angular-socialshare#100]
For some providers such as facebook, sometimes it will not work if you do not provide a redirect_uri (like the send dialog), it would be nice to be able to retrieve the reference of `$window` that was open so that we can close it from the source (maybe by storing in the `$scope.socialShareWindow` ?)

**Comment by 45kb** [comment]
Actually i did not try this, but as far as i remember, javascript doesn't allow cross origins references/interactions, maybe i am mistaking, but the only way could be to use https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage which looks not fully supported.

Otherwise you will not be able to know when the popup has really finished the process of sharing, i guess/bet.

Do you got any idea?

**Comment by mebibou** [comment]
Only the window that opens the popup window can close it, which is fine because the idea is to be able to close it and that's all. If someone needs to know when to close it, that's a different problem yes.

One could actually easily access the parent window from the popup window (which is allowed) and call a method that would close the popup, since the parent window has the reference, so no problem here.

**Comment by 45kb** [comment]
That's clear, what i meant is, you would close the popup AFTER user shared the content, and my doubt is, how you fire when it's shared...

**Comment by mebibou** [comment]
With facebook you can use `redirect_uri` and display a link that would say "Click here to close this window". For others I don't know, I think some might close it automatically, but you cannot control what each providers does so I'm not sure why you want to solve this.

**Comment by 45kb** [comment]
Yes indeed, with a redirect-uri everything is possible, if we can't intercept it for any of all the other providers then it doesn't worth a try,  imho.
